### PR TITLE
[COST-2561] resource level gcp downloader

### PR DESCRIPTION
**Pull request by adberglund**
Fixes [COST-2561](https://issues.redhat.com/browse/COST-2561)

Changes proposed in this PR:
* This will pull resource level fields in the GCP Downloader when they are available
* NULL is inserted when the underlying table is not a resource-level table.

Testing Instructions:
1. Run `this command` ...
2. Ingest this data `with this other command` ...
3. Expect to see this result:
   ```
   GET http://localhost:8000/api/endpoint

   {json: response}
   ```
4.
5.

---
Additional Context:

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
8655a8af-bb5f-41d6-8318-4fee23226ca2
a996c1d6-c47e-4188-b872-8d6db46c3f86

File: koku/masu/external/downloader/gcp/gcp_report_downloader.py [hunk 1]
```diff
@@ -345,6 +348,16 @@ def build_query_select_statement(self):
             f"TO_JSON_STRING({col})" if col in ("labels", "system_labels", "project.labels") else col
             for col in columns_list
         ]
+        # Swap out resource columns wit NULLs when we are processing
```
Comment (maskarb) [inline koku/masu/external/downloader/gcp/gcp_report_downloader.py:351]: nit-pick:
```suggestion
        # Swap out resource columns with NULLs when we are processing
```

File: koku/masu/external/downloader/gcp/gcp_report_downloader.py [hunk 2]
```diff
@@ -345,6 +348,16 @@ def build_query_select_statement(self):
...
+        # a non-resource-level BigQuery table
+        columns_list = [
+            f"NULL as {col.replace('.', '_')}"
+            if col in ("resource.name", "resource.global_name")
+            and RESOURCE_LEVEL_EXPORT_NAME not in self.data_source.get("table_id")
+            else col
+            for col in columns_list
+        ]
+        # columns_list = self.modify_query_select_for_resource_columns(columns_list)
```
Comment (maskarb) [inline koku/masu/external/downloader/gcp/gcp_report_downloader.py:360]: nit-pick:
```suggestion
```

**Review by maskarb — APPROVED** [review]
Looks good to me! Just a couple of nit-picks:

**Pull request merged**